static/app/components/modals/inviteMembersModal/inviteRowControlNew.tsx

```diff
@@ -0,0 +1,227 @@
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import type {StylesConfig} from 'sentry/components/forms/controls/selectControl';
+import SelectControl from 'sentry/components/forms/controls/selectControl';
+import {useInviteMembersContext} from 'sentry/components/modals/inviteMembersModal/inviteMembersContext';
+import RoleSelectControl from 'sentry/components/roleSelectControl';
+import TeamSelector from 'sentry/components/teamSelector';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {SelectValue} from 'sentry/types/core';
+import type {OrgRole} from 'sentry/types/organization';
+
+import renderEmailValue from './renderEmailValue';
+import type {InviteStatus} from './types';
+
+type SelectOption = SelectValue<string>;
+
+type Props = {
+  roleDisabledUnallowed: boolean;
+  roleOptions: OrgRole[];
+};
+
+function ValueComponent(
+  props: MultiValueProps<SelectOption>,
+  inviteStatus: InviteStatus
+) {
+  return renderEmailValue(inviteStatus[props.data.value], props);
+}
+
+function mapToOptions(values: string[]): SelectOption[] {
+  return values.map(value => ({value, label: value}));
+}
+
+function InviteRowControl({roleDisabledUnallowed, roleOptions}: Props) {
+  const {inviteStatus, pendingInvites, setEmails, setRole, setTeams, reset} =
+    useInviteMembersContext();
+  const emails = [...(pendingInvites.emails ?? [])];
+  const role = pendingInvites.role ?? '';
+  const teams = [...(pendingInvites.teams ?? [])];
+
+  const onChangeEmails = (opts: SelectOption[]) => {
+    setEmails(opts?.map(v => v.value) ?? [], 0);
+  };
+  const onChangeRole = (value: SelectOption) => setRole(value?.value, 0);
+  const onChangeTeams = (opts: SelectOption[]) =>
+    setTeams(opts ? opts.map(v => v.value) : [], 0);
+
+  const [inputValue, setInputValue] = useState('');
+
+  const theme = useTheme();
+
+  const isTeamRolesAllowedForRole = useCallback<(roleId: string) => boolean>(
+    roleId => {
+      const roleOptionsMap = roleOptions.reduce(
+        (rolesMap, roleOption) => ({...rolesMap, [roleOption.id]: roleOption}),
+        {}
+      );
+      return roleOptionsMap[roleId]?.isTeamRolesAllowed ?? true;
+    },
+    [roleOptions]
+  );
+  const isTeamRolesAllowed = isTeamRolesAllowedForRole(role);
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
+    switch (e.key) {
+      case 'Enter':
+      case ',':
+      case ' ':
+        e.preventDefault();
+        handleInput(inputValue);
+        setInputValue('');
+        break;
+      default:
+      // do nothing.
+    }
+  };
+
+  const handleInput = input => {
+    const newEmails = input.trim() ? input.trim().split(/[\s,]+/) : [];
+    if (newEmails.length > 0) {
+      onChangeEmails([
+        ...mapToOptions(emails),
+        ...newEmails.map(email => ({label: email, value: email})),
+      ]);
+    }
+  };
+
+  return (
+    <RowWrapper>
+      <div>
+        <Heading>Email addresses</Heading>
```

Comment:
These heading elements should be `label`'s instead of divs, and then you can use `htmlFor` to connect the label to the control. Then the control wouldn't need the `aria-label` anymore.

https://medium.com/@amirakhaled2027/decoding-the-mystery-of-htmlfor-label-association-db775a61f13e

Reply:
I think we still need `aria-label` in order to write tests, for example
`screen.getByRole('textbox', {name: 'Email Addresses'})`

If I leave out the `aria-label` it looks like the `textbox` doesn't have a name? Not sure if I'm missing something